http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/DiskRoutes.scala

```diff
@@ -0,0 +1,316 @@
+      context <- AppContext.generate[IO]
+      implicit0(ctx: ApplicativeAsk[IO, AppContext]) = ApplicativeAsk.const[IO, AppContext](
+        context
+      )
+      _ <- diskService.updateDisk(userInfo, googleProject, diskName, req)
+    } yield StatusCodes.Accepted
+}
+
+object DiskRoutes {
+
+  implicit val createDiskRequestDecoder: Decoder[CreateDiskRequest] = Decoder.instance { c =>
+    for {
+      l <- c.downField("labels").as[Option[LabelMap]]
+      s <- c.downField("size").as[Option[DiskSize]]
+      t <- c.downField("diskType").as[Option[DiskType]]
+      bs <- c.downField("blockSize").as[Option[BlockSize]]
+    } yield CreateDiskRequest(
+      l.getOrElse(Map.empty),
+      s,
+      t,
+      bs
+    )
+  }
+
+  implicit val updateDiskRequestDecoder: Decoder[UpdateDiskRequest] = Decoder.instance { x =>
+    for {
+      l <- x.downField("labels").as[LabelMap]
+      us <- x.downField("updateSize").as[Option[DiskSize]]
+      ud <- x.downField("updateDiskType").as[Option[DiskType]]
+      ub <- x.downField("updateBlockSize").as[Option[BlockSize]]
+    } yield UpdateDiskRequest(l, us, ud, ub)
+  }
+
+  implicit val getDiskResponseEncoder: Encoder[GetPersistentDiskResponse] = Encoder.forProduct12(
+    "id",
+    "googleProject",
+    "zone",
+    "name",
+    "googleId",
+    "samResourceId",
+    "status",
+    "auditInfo",
+    "size",
+    "diskType",
+    "blockSize",
+    "labels"
+  )(x =>
+    (
+      x.id,
+      x.googleProject,
+      x.zone,
+      x.name,
+      x.googleId,
+      x.samResourceId,
+      x.status,
+      x.auditInfo,
+      x.size,
+      x.diskType,
+      x.blockSize,
+      x.labels
+    )
+  )
+
+  implicit val listDiskResponseEncoder: Encoder[ListPersistentDiskResponse] = Encoder.forProduct12(
+    "id",
+    "googleProject",
+    "zone",
+    "name",
+    "googleId",
+    "samResourceId",
+    "status",
+    "auditInfo",
+    "size",
+    "diskType",
+    "blockSize",
+    "labels"
+  )(x =>
+    (
+      x.id,
+      x.googleProject,
+      x.zone,
+      x.name,
+      x.googleId,
+      x.samResourceId,
+      x.status,
+      x.auditInfo,
+      x.size,
+      x.diskType,
+      x.blockSize,
+      x.labels
+    )
+  )
+
+  private val diskNameReg = "([a-z|0-9|-])*".r
+
+  private def validateDiskName(diskNameString: String): Either[Throwable, DiskName] =
+    diskNameString match {
+      case diskNameReg(_) => Right(DiskName(diskNameString))
+      case _ =>
+        Left(
+          RequestValidationError(
+            s"invalid disk name ${diskNameString}. Only lowercase alphanumeric characters, numbers and dashes are allowed in disk name"
+          )
+        )
+    }
+
+  def validateDiskNameDirective(diskNameString: String): Directive1[DiskName] =
+    Directive { inner =>
+      validateDiskName(diskNameString) match {
+        case Left(e)  => failWith(e)
+        case Right(c) => inner(Tuple1(c))
+      }
+    }
+
+}
+
+final case class ListPersistentDiskResponse(id: DiskId,
```

Comment:
As of the moment, these are the same as the `PersistentDisk` type. Once we include `errors` in PDs, I think that will go in the `get` response and not the `list` response. Other potential fields we could remove from the `list` response are `googleId`, `samResourceId`, and `labels` - let me know what you think

Reply:
sounds good to me...are you removing the fields you mentioned? or you're just saying that can be done in the future? (I'd prefer lets just remove them if no particular reason to keep)